In [2]:
pip install pymatgen ase matplotlib pandas numpy jupyter

Note: you may need to restart the kernel to use updated packages.


In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

import pymatgen as mg
from pymatgen.core import Structure
from ase import Atoms
from ase.visualize import view

In [8]:
structure1 = Structure.from_file("..\\data\\sillicon.cif")

print(structure1)

Full Formula (Si8)
Reduced Formula: Si
abc   :   5.430700   5.430700   5.430700
angles:  90.000000  90.000000  90.000000
pbc   :       True       True       True
Sites (8)
  #  SP       a     b     c
---  ----  ----  ----  ----
  0  Si    0     0     0
  1  Si    0     0.5   0.5
  2  Si    0.5   0     0.5
  3  Si    0.5   0.5   0
  4  Si    0.75  0.75  0.25
  5  Si    0.75  0.25  0.75
  6  Si    0.25  0.75  0.75
  7  Si    0.25  0.25  0.25


In [6]:
# load another structure

structure2 = Structure.from_file("..\\data\\h-BN.cif")

print(structure2)

Full Formula (B2 N2)
Reduced Formula: BN
abc   :   2.510000   2.510000   6.690000
angles:  90.000000  90.000000 120.000000
pbc   :       True       True       True
Sites (4)
  #  SP           a         b     c
---  ----  --------  --------  ----
  0  B3+   0.333333  0.666667  0.05
  1  B3+   0.666667  0.333333  0.95
  2  N3-   0         0         0
  3  N3-   0         0         0.5


c:\Users\Admin\anaconda3\Lib\site-packages\pymatgen\core\structure.py:3112: UserWarning: Issues encountered while parsing CIF: 2 fractional coordinates rounded to ideal values to avoid issues with finite precision.
  struct = parser.parse_structures(primitive=primitive)[0]


### 1. h-BN exbibit hexagonal symmetry, while silicon cubic symmetry
### 2. silicon has equal lattice constant along three axes, indicating a highly symmetric cubic structure
### 3. The gamma angle of 120 degree in h-BN is characteristics of hexagonal crystal systems.
### 4. h-BN contains two different atomic species B and N, whereas silicon is composed of a single element